src/components/Grid/Pagination.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import styles from './Pagination.module.css'
+
+const usePagination = ({
+  data = null,
+  total = null,
+  per_page = null,
+  total_pages = null,
+  current_page = 1,
+  fetchPage = null,
+  startPageNumber = 1,
+}) => {
+  if (!fetchPage) {
+    throw Error('usePagination requires a fetchPage callback')
+  }
+
+  const [pagingState, setPagingState] = useState({
+    data,
+    total,
+    per_page,
+    total_pages,
+    current_page,
+    fetchPage,
+  })
+
+  const fetchPageDelegate = async (pageNumber) => {
+    const data = await fetchPage(pageNumber)
```

Comment:
I might be late to the discussion,
but if the pagination experience lets the user choose the per page item count, that should also be part of the data sent to back-end, in addition to `pageNumber`.

Also, if there is a global sort (i.e., sort all the response data based on, say date) we need to pass in the `sortField` and `sortOrder` to the backend.
This is a longer discussion, but just putting my thoughts for the record.

Reply:
@GreyFalcon isn't `per_page` supposed to be the item count?

Reply:
+1 On frontend providing the per page count.

Reply:
@roblevintennis what is the plan for backend vs frontend sorting? Is it something that Tom needs to weigh in?

Reply:
Just asked in our Slack channel about the two points above. I don't see a huge use case for building in the per page item count, but since I know that sorting was already something that Tom has in his mocks, `sortField` and `sortOrder` makes sense

Reply:
> lets the user choose the per page item count,

Excellent feedback Ashwin!

I think for MVP we are not letting the user choose 10/20/50, but, the API should certainly allow for it I agree! It's hard to imagine we won't eventually offer that. Personally, I use that to enlarge the count per page whenever possible -- so I'm a fan!

Ultimately, I believe we should offer `page` and `per_page` so I agree.

From lack of push-back, I'm taking it that the format I've been asking for is fine with all. To be repetitive, it's something llike:

...json
page": 2,
    "per_page": 6,
    "total": 12,
    "total_pages": 2,
    "data": [...]
...
```

___

> what is the plan for backend vs frontend sorting

I think we can certainly be iterative, but, my first assumption would be:

- field sorting is passed in `field=direction` so `sortField` and `sortOrder` only need to be one query param: `?status=desc` 
- per Emma's input, `date=desc` will always be implied such that:
  1. default with no sort asked for will be sorted by date most recent at top aka
> Reverse chronological order would list them from latest to earliest.
  2. if we've asked for `?status=desc` that would be the primary sort, and if there are duplicates status, those will be secondarily ordered in reverse chronological ordering such that the most recent ones be listed before earlier ones